### ISSUE TITLE
GM Note do not work with ApplicationV2

### ISSUE THREAD

**Issue by bithir** [bithir/gm-notes#39]

**Comment by bithir** [comment]
Header controls can now be customized for ApplicationV2 with a new hook signature getHeaderControls{ApplicationClassName}(app, controls).

https://github.com/foundryvtt/foundryvtt/issues/11668

**Comment by ullantra** [comment]
Hi, I assume this is the issue why we can't see the header button to actually open the notes in V13/DND5e 5.0.x?

**Comment by bithir** [comment]
Yep - it is related to AppV2 changes and the removal of jQuery.

**Comment by Lertuga** [comment]
Hello,
I am having the same issue with the module. I just migrated from past versions to V13 and was wondering if the issue will be solvable in the future? Just to know if I should go back to past version

Thank you for your time and the amazing module!

**Comment by Lertuga** [comment]
> Fixed, but I do not run a system with AppV2 actor sheets, item sheets. So if someone could validate that works in 10.3 for me, that be great.

Unfortunately I am not the right person to assist you with that, as I don't know what that means. However, I checked with version 10.4 and it still doesn't have the Notes button.

**Comment by bithir** [comment]
Note, that in AppV2, the notes button can only appear in the window sub-menu.

<img width="295" height="112" alt="Image" src="https://github.com/user-attachments/assets/de0d48a7-24bf-4360-ad63-532cc8f447d5" />

Is that where you are looking?  And with what system and type (actor/item/tile/light/wall/etc) are you looking?

**Comment by Lertuga** [comment]
Yes, I was also looking in that sub-menu, the only option is Configure Sheet.

I am checking tath for an "Item" (I just tried with "Actors" and the Notes button is in the sub-menu.

I am using Version 13 Build 346, Dnd5e 5.0.4 and the module is 10.4



> Note, that in AppV2, the notes button can only appear in the window sub-menu.
> <img alt="Image" width="295" height="112" src="https://private-user-images.githubusercontent.com/19734037/467392489-de0d48a7-24bf-4360-ad63-532cc8f447d5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI3NTI2MDQsIm5iZiI6MTc1Mjc1MjMwNCwicGF0aCI6Ii8xOTczNDAzNy80NjczOTI0ODktZGUwZDQ4YTctMjRiZi00MzYwLWFkNjMtNTMyY2M4ZjQ0N2Q1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzE3VDExMzgyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiZTYyNmVmYWZkOTZkNDNlMTk1OTFmYzU3YjEwOTIzODc5MzY1NTc2Mzc4ZWMwN2E0Yzg1YTQyOTUzMzM0MjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3u6sW4qaTGKYouE93TlpVki89KDaP9c4tbWXBrb9niE">
> 
> Is that where you are looking? And with what system and type (actor/item/tile/light/wall/etc) are you looking?

**Comment by bithir** [comment]
Hmmm

<img width="1301" height="327" alt="Image" src="https://github.com/user-attachments/assets/e976925e-fb4c-46de-baf6-4072091eb0b2" />

**Comment by bithir** [comment]
Hmmm....
ItemSheet behaviour seems poor in 5e. Unsure on why.  It never triggers getHeaderControlsItemSheetV2 as I would have anticipated.

Evidence ItemSheet hook list:

<img width="1420" height="797" alt="Image" src="https://github.com/user-attachments/assets/e7e9d39f-c088-4958-b212-5cb3627f701a" />

ActorSheet hook list:

<img width="1216" height="896" alt="Image" src="https://github.com/user-attachments/assets/5999e700-3d1e-46f9-9e92-a2ccd32fca2f" />

Note that the getHeaderControlsActorSheetV2 is called correctly for actors, but getHeaderControlsItemSheetV2 is not for items.

Something perhaps to raise with Atropos - unsure if it is a bug in Foundry VTT v13 or specific to dnd5e

**Comment by bithir** [comment]
https://discord.com/channels/170995199584108546/1243227783082283140/1395400194169901087

**Comment by bithir** [comment]
<img width="1421" height="1060" alt="Image" src="https://github.com/user-attachments/assets/e5159bb1-a656-4b12-a538-4c19e1478a6d" />

**Comment by bithir** [comment]
So, we need to revise the hook logic to make 5e work.
